### ISSUE TITLE
Allow ['catch'] as a synonym for .catch

### ISSUE THREAD

**Issue by tylerbrandt** [xjamundx/eslint-plugin-promise#65]
If code has to support being parsed by IE8, it is necessary to avoid using the reserved word "catch" as an identifier, i.e.:
```js
new Promise(function(resolve, reject) {
  ...
})
.then(...)
['catch'](function(err) { ... })
```
See [this note](https://github.com/stefanpenner/es6-promise#usage-in-ie9) in es6-promise for details.

Unfortunately, this pattern triggers the "promise/catch-or-return" rule, so one must consistently disable that rule to avoid having warnings. It would be ideal if `['catch']` would be interpreted the same as `.catch`. Is that possible, or is there a workaround (without using a transpiler to rewrite the code)?

**Comment by xjamundx** [comment]
I think there's already an eslint rule for this. I can check!

**Comment by xjamundx** [comment]
Sorry I had misinterpreted this request. This should be fairly straight forward to support.

**Comment by xjamundx** [comment]
I'll try and work on this today if I can

**Comment by xjamundx** [comment]
Published in `3.6.0`
